### PR TITLE
Prepublishing Nudges : Publish scheduling

### LINKED ISSUE #11607: Prepublishing Nudges : Add Publish scheduling option
1. When the user tap "Publish"
2. Then show the DatePicker and the user can schedule when the post will be published.

### PR DESCRIPTION
**Pull request by jd-alexander**
Fixes #11607

## Solution
1. `PublishSettingsFragment` & `PublishSettingsViewModel` from #11727 were used to create `PrepublishingPublishSettingsFragment` & `PrepublishingPublishSettingsViewModel`. 

2. The Date/Time picker & notification dialogs were using a reference to the `EditPostPublishSettingsViewModel` to update the selected dates. This was refactored to select the right `ViewModel` type depending on the screen that is using the functionality. I was going to use some form of `.class::java` approach, by passing class types from the calling fragment but decided not to and opted for enums instead 933be4a9646dad1327207e0007c5495ac1fbf948 2d790d7

## Testing
1. To test simply create a new post. 
2. Press publish. 
3. Once the bottom sheet is shown click the publish scheduling button, and you will see the `Publish` screen with the date picking mechanisms. 
4. Change the date and then go back to the home of the bottom sheet to see the changes.
5.Go to Post Settings.
6. Click on Publish.
7. Ensure you see the values that were set in the bottom sheet.

Publish immediately | Scheduled | Home screen 
--------|------- |-------
   |  |   

------------
Since the `EditPostPublishSettings` viewmodel & fragment logic was modified in this PR, it would be good to a test to ensure the functionality is still intact. 

1. Create a post.
2. Go to Post Settings.
3. Click on Publish.
4. Change the Time & Date.
5. Also, change the Notification.
6. Go back and ensure the values have been set and reflected in the Post Settings UI.

## Reviewing 

1. Only 1 reviewer is needed but anyone can review.
2. Ensure that #11727 is merged.
3. Set this base branch to feature/prepublishing_nudges
4. Remove Not Ready for Merge
5. Merge :)

## Submitter Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [x] I have considered adding accessibility improvements for my changes.
- [x] If it's feasible, I have added unit tests.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsFragment.kt [hunk 1]
```diff
@@ -38,159 +19,6 @@ class EditPostPublishSettingsFragment : Fragment() {
                 .get(EditPostPublishSettingsViewModel::class.java)
     }
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsFragment.kt:41]: Could you please create a separate PR where you just create the parent fragment and move the logic away from this fragment? It's very difficult to check what you just moved vs changed. Thank you ! ;)

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt [hunk 2]
```diff
@@ -1,271 +1,21 @@
...
 import org.wordpress.android.fluxc.store.SiteStore
-import org.wordpress.android.ui.posts.EditPostRepository.UpdatePostResult
-import org.wordpress.android.util.DateTimeUtils
 import org.wordpress.android.util.LocaleManagerWrapper
-import org.wordpress.android.viewmodel.Event
 import org.wordpress.android.viewmodel.ResourceProvider
-import java.util.Calendar
 import javax.inject.Inject
 
-class EditPostPublishSettingsViewModel
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt:28]: This would be also great to do in a separate PR, thank you!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsFragment.kt (hunk 1 above)
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsFragment.kt]: Yes, I did that here :) https://github.com/wordpress-mobile/WordPress-Android/pull/11728

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt (hunk 2 above)
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt]: Yes, I did that here :) https://github.com/wordpress-mobile/WordPress-Android/pull/11728

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostDatePickerDialogFragment.kt [hunk 3]
```diff
@@ -10,17 +10,29 @@ import androidx.lifecycle.ViewModelProvider
...
+    private lateinit var viewModel: PublishSettingsViewModel
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
-        viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
-                .get(EditPostPublishSettingsViewModel::class.java)
+        val publishSettingsFragmentType = arguments?.getParcelable<PublishSettingsFragmentType>(
+                ARG_PUBLISH_SETTINGS_FRAGMENT_TYPE
+        )
+
+        when (publishSettingsFragmentType) {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostDatePickerDialogFragment.kt:27]: Good call with the enum! ;)

**Comment by jd-alexander** [comment]
> Wow, wonderful job @jd-alexander! Both the code and the app looks great ;)! Thanks for breaking this PR down into smaller chunks. It was way easier to review :).

Thanks for the review again @malinajirka 🙏 

>SideNote:
> I have noticed that the bottom sheet is shown only for the "Publish" action. When I for example change the schedule time in post settings the action changes "Publish -> Schedule" and the bottom sheet doesn't show up. I assume this is not related to this PR and it's a work to be done later.

Oh yes, I had put this in the project board as an issue to check out. I think it has to do with how the `primaryActions` in the `EditPostActivity` operates. 

> I'm not sure about the design of this row

Yes, this is intended. The home screen UI is being redesigned in another PR as we speak.
https://github.com/wordpress-mobile/WordPress-Android/issues/11697

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostNotificationScheduleTimeDialogFragment.kt [hunk 4]
```diff
@@ -15,15 +15,27 @@ import org.wordpress.android.fluxc.store.PostSchedulingNotificationStore.Schedul
...
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
-        viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
-                .get(EditPostPublishSettingsViewModel::class.java)
+        val publishSettingsFragmentType = arguments?.getParcelable<PublishSettingsFragmentType>(
+                ARG_PUBLISH_SETTINGS_FRAGMENT_TYPE
+        )
+
+        when (publishSettingsFragmentType) {
+            EDIT_POST -> viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
```
[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PostNotificationScheduleTimeDialogFragment.kt:33]: Yes, we can. I wanted to change out all uses of this but wasn't sure if this change would be unrelated to this PR, so I reverted them here 938e599 haha but I will put them back 🙏

[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PostNotificationScheduleTimeDialogFragment.kt:33]: Tracking this here : https://github.com/wordpress-mobile/WordPress-Android/issues/11751
